app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -212,21 +159,28 @@ private fun ConversationScreenContent(
     onSendAttachment: (AttachmentBundle?) -> Unit,
     onDownloadAsset: (String) -> Unit,
     onImageFullScreenMode: (String) -> Unit,
-    onMessageComposerError: (ConversationErrors) -> Unit,
+    onMessageComposerError: (ConversationSnackbarMessages) -> Unit,
     conversationState: ConversationViewState,
     conversationScreenState: ConversationScreenState
 ) {
     val lazyListState = rememberLazyListState()
     val coroutineScope = rememberCoroutineScope()
+    val context = LocalContext.current
 
     LaunchedEffect(conversationState.messages) {
         lazyListState.animateScrollToItem(0)
     }
 
-    conversationState.onError?.let { errorCode ->
-        val errorMessage = getErrorMessage(errorCode)
-        LaunchedEffect(conversationState.onError) {
-            conversationScreenState.snackBarHostState.showSnackbar(errorMessage)
+    conversationState.onSnackbarMessage?.let { msgCode ->
+        val (message, actionLabel) = getSnackbarMessage(msgCode)
+        LaunchedEffect(conversationState.onSnackbarMessage) {
+            val snackbarResult = conversationScreenState.snackBarHostState.showSnackbar(message = message, actionLabel = actionLabel)
+            when {
+                // Show downloads folder when clicking on Snackbar cta button
+                msgCode is OnFileDownloaded && snackbarResult == SnackbarResult.ActionPerformed -> {
+                    context.startActivity(Intent(DownloadManager.ACTION_VIEW_DOWNLOADS))
+                }
```

Comment:
you could clear the onSnackbarMessage here instead inside the ViewModel, that way you are certan that before clearing it showSnackbar method is called

Reply:
You mean after having a delay to show it correctly, calling a method from the VM to reset it to null? I feel there is no harm if that logic lives directly in the VM, instead of bloating more the UI part.

Reply:
in my. opinion it fits it better inside the UI, because imagine the sceneratio, that we set the onSnackbarMessage to a msgCode, but the user opens another app in the mean time and goes back to Wire since, this is a coroutine it will continue executing and the snackbar message will never appear, when you do it in the UI, you are sure that it will be shown and once it is shown we reset the onSnackbarMessage to null, correct me if I am wrong on it of course

Reply:
also you wouldn't need this magic delay